scrnaseq.Rmd

```diff
@@ -1548,14 +1691,17 @@ if ((known_markers_n > 0) & any(idx_dotplot)) {
   for (i in seq(known_markers_dotplot)) {
     g = known_markers_dotplot[[i]]
     g = g[length(g):1]
-    p_list[[i]] = suppressMessages(Seurat::DotPlot(sc, features=g) + 
-      scale_colour_gradient2(low="steelblue", mid="lightgrey", high="darkgoldenrod1") +
-      AddStyle(title=paste("Known marker genes:", names(known_markers_dotplot)[i]), 
-               ylab="Cluster") + 
-      theme(axis.text.x=element_text(angle=90, hjust=1, vjust=.5)))
+    p_list[[i]] = suppressMessages(
+      Seurat::DotPlot(sc, features=g) + 
+        scale_colour_gradient2(low="steelblue", mid="lightgrey", high="darkgoldenrod1") +
+        AddStyle(title=paste("Known marker genes:", names(known_markers_dotplot)[i]), ylab="Cluster") + 
+        theme(axis.text.x=element_text(angle=90, hjust=1, vjust=.5)) +
+        lims(size=c(0,100)) +
+        guides(color = FALSE)
+      )
   }
-  p = patchwork::wrap_plots(p_list, ncol=1)
-  print(p)
+  p = patchwork::wrap_plots(p_list, ncol=1) + patchwork::plot_layout(guides="collect")
```

Comment:
I'd prefer to have the scales next to each plot.

Reply:
done